core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -252,15 +267,36 @@ private[spark] class BlockManager(
 
     blockManagerId = if (idFromMaster != null) idFromMaster else id
 
-    shuffleServerId = if (externalShuffleServiceEnabled) {
+    if (externalk8sShuffleServiceEnabled) {
+      remoteShuffleServiceAddress = mapOutputTracker
+        .trackerEndpoint
+        .askSync[List[(String, Int)]](GetRemoteShuffleServiceAddresses)
+    }
+
+    shuffleServerId = if (externalk8sShuffleServiceEnabled) {
+      // TODO: Investigate better methods of load balancing
+      // note: might break if re-initialized
+      randomShuffleServiceAddress = remoteShuffleServiceAddress.head
```

Comment:
What's stopping us from just doing `remoteShuffleServiceAddress.shuffle.head`? Picking a random one arbitrarily shouldn't be any different than picking the first one every time.